scipy/spatial/transform/rotation.py

```diff
@@ -1461,6 +1464,89 @@ def magnitude(self):
+
+        Returns
+        -------
+        reduced : `Rotation` instance
+            Object containing reduced rotations.
+        """
+        if left is None and right is None:
+            raise ValueError("No reduction groups provided")
+        elif right is None:
+            q = left
+        elif left is None:
+            q = right
+        else:
+            q = [(e * right).as_quat() for e in left]
+            q = Rotation.from_quat(np.vstack(q))
+
+        p = self
+        A = p.as_quat().reshape((len(p), 4))
+        B = q.as_quat().reshape((len(q), 4))
+
+        cosines = np.outer(A[:, 3], B[:, 3])
+        cosines -= np.outer(A[:, 0], B[:, 0])
+        cosines -= np.outer(A[:, 1], B[:, 1])
+        cosines -= np.outer(A[:, 2], B[:, 2])
+        cosines = np.minimum(1, np.abs(cosines))
+        """
+        We want to compute:
+        `closest = [np.argmin((p * q).magnitude()) for r in p]`
+        A cheaper way of obtaining this is:
+        `closest = [np.argmax(abs((p * q)[:, 3])) for r in p]`
+        Since we are only interested in the `w` component, we do not have to
+        compute a full quaternion multiplication.  The outer product above
+        calculates all `w` components simultaneously.  We can then find the
+        index of the closest rotation using:
+        `np.argmax(cosines, axis=1)`
+        """
+        return p * q[np.argmax(cosines, axis=1)]
+
+    @classmethod
+    def create_group(cls, group, axis='Z'):
+        """Create a 3D rotation group.
+
+        Parameters
+        ----------
+        group: string
+            The name of the group. Must be one of 'I', 'O', 'T', 'Dn', 'Cn',
+            where `n` is a positive integer. The groups are:
+                I: Icosahedral group. Fixes the rotations of an icosahedron.
+                O: Octahedral group. Fixes the rotations of an octahedron.
+                T: Tetrahedral group. Fixes the rotations of a tetrahedron.
+                D: Dicyclic group. Fixes the rotations of an n-sided prism.
+                C: Cyclic group. Fixes the rotations of an n-sided pyramid.
+        axis: integer
+            The cyclic rotation axis. Must be one of ['X', 'Y', 'Z'] (or
+            lowercase). Default is 'Z'. Ignored for groups 'I', 'O', and 'T'.
+
+        Returns
+        -------
+        rotation : `Rotation` instance
+            Object containing the elements of the rotation group.
+
+        Notes
+        -----
+        This method generates rotation groups only. The full 3-dimensional
+        groups also contain reflections. See e.g. [Conway]_ for a good
+        description of the rotation groups and full 3-dimensional groups.
+
+        References
+        ----------
+        .. [Conway] John H Conway and Derek A Smith, On quaternions and
+                    octonions, 2003, AK Peters/CRC Press, ISBN 978-1568811345
+        """
+        return scipy.spatial.transform._rotation_groups._create_group(cls, group, axis=axis)
```

Comment:
I think you can and should just import `create_group`.

Reply:
Fixed.